agent/agent_worker.go

```diff
@@ -647,10 +647,15 @@ func (a *AgentWorker) RunJob(ctx context.Context, acceptResponse *api.Job) error
 	})
 
 	// Now that we've got a job to do, we can start it.
-	jr, err := NewJobRunner(a.logger, jobMetricsScope, a.agent, acceptResponse, a.apiClient, JobRunnerConfig{
+	jr, err := NewJobRunner(JobRunnerConfig{
+		Job:                acceptResponse,
+		APIClient:          a.apiClient,
+		Logger:             a.logger,
```

Comment:
I'm not a fan of having these in the config. IMO, config should be serialisable. It is a lot easier to test functions with serialisable inputs and outputs.

I get that `a.agentStdout` isn't serialisable either, but we shouldn't double down. Rather, it would be great if you could separate the serialisable and not serialisable parts.

Reply:
yeah, i wavered a little mid implementation thinking the same thing, and tentatively came down on the other side, but i think you're right.

is your preferred signature
```Go
NewThingo(a NonSerializable, b NonSerializable, cfg ThingoConfig)
```